pyper/core/stage.py

```diff
@@ -1,8 +1,24 @@
-from typing import List, Any
+from enum import Enum
+from typing import List, Any, Callable, Union
 
 from pydantic.fields import Field
 from pydantic.main import BaseModel
 
+from pyper.task import Task
+
+
+class StageStatus(str, Enum):
+    NOT_STARTED = "not_started"
+    STARTED = "started"
+    ERROR = "error"
+    DONE = "done"
+
 
 class Stage(BaseModel):
-    result: List[Any] = Field(default_factory=[])
+    result: List[Any] = Field(default_factory=list)
+    status: StageStatus = StageStatus.NOT_STARTED
+    task: Task = None
```

Comment:
```suggestion
    last_task: Task = None
```
Apparently, it seems like this field indicates the last active task instead of single task. How about change it to `last_task` or `last_active_task`? IMHO this makes what Stage class does clear.

Reply:
agreed too.

Reply:
agreed too.